app/browser/reducers/downloadsReducer.js

```diff
@@ -89,6 +92,18 @@ const downloadsReducer = (state, action) => {
         state = state.set('downloads', downloads)
       }
       break
+    case appConstants.APP_DOWNLOAD_DEFAULT_PATH:
+      const focusedWindow = BrowserWindow.getFocusedWindow()
+
+      dialog.showOpenDialog(focusedWindow, {
+        defaultPath: app.getPath('downloads'),
+        properties: ['openDirectory']
+      }, (folder) => {
+        if (Array.isArray(folder) && fs.lstatSync(folder[0]).isDirectory()) {
+          appActions.changeSetting(settings.DOWNLOAD_DEFAULT_PATH, folder[0])
```

Comment:
I think this is fine as-is... but it could be better as it's own discreet action (in the event that we'd like other reducers to be listening).  Perhaps an action like `defaultDownloadPathChanged`